app/javascript/lib/validation/expression-matrices-validation.js

```diff
@@ -62,9 +72,12 @@ export async function parseBarcodesFile(chunker, mimeType, fileOptions) {
   let issues = []
 
   const dataObj = {} // object to track multi-line validation concerns
-  await chunker.iterateLines((rawLine, lineNum, isLastLine) => {
-    issues = issues.concat(validateUniqueRowValuesWithinFile(rawLine, isLastLine, dataObj))
+  await chunker.iterateLines({
+    func: (rawLine, lineNum, isLastLine) => {
+      issues = issues.concat(timeOutCSFV(chunker))
```

Comment:
```suggestion
      if (lineNum % 100 === 0) {
        issues = issues.concat(timeOutCSFV(chunker))
      }
```

Reply:
I like the idea of not having the time check on every line, but this is going to defeat the purpose of the time limit on files with really long lines, or a misformatted file that doesn't have any newlines. Also, the time check operation should be much, much faster than actually parsing a line, so I don't think this will materially impact performace

Reply:
See [reply](https://github.com/broadinstitute/single_cell_portal_core/pull/1366#discussion_r803849935).